### PR TITLE
Fixed issues in uninstall function of AppM intergration

### PR DESCRIPTION
**Pull request by milanperera**

### CODE REVIEW

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 1]
```diff
@@ -54,960 +54,975 @@
...
+	private static final int PRE_WIPE_WAIT_TIME = 4000;
+	private static final int ACTIVATION_REQUEST = 47;
+	private static final int DEFAULT_PASSWORD_LENGTH = 0;
+	private static final int DEFAULT_VOLUME = 0;
+	private static final int DEFAULT_FLAG = 0;
+	private static final int DEFAULT_PASSWORD_MIN_LENGTH = 3;
+	private static final long DAY_MILLISECONDS_MULTIPLIER = 24 * 60 * 60 * 1000;
+	private Map<String, String> bundleParams;
+
+	public Operation(Context context) {
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 2]
```diff
@@ -54,960 +54,975 @@
...
+		gps = new GPSTracker(context.getApplicationContext());
+	}
+
+	/**
+	 * Executes device management operations on the device.
+	 *
+	 * @param operation - Operation object.
+	 */
+	public void doTask(org.wso2.mdm.agent.beans.Operation operation) throws AndroidAgentException {
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 3]
```diff
@@ -54,960 +54,975 @@
...
+		}
+	}
+
+	/**
+	 * Lock the device.
+	 *
+	 * @param operation - Operation object.
+	 */
+	public void lockDevice(org.wso2.mdm.agent.beans.Operation operation) {
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 4]
```diff
@@ -54,960 +54,975 @@
...
+		}
+	}
+
+	/**
+	 * Ring the device.
+	 *
+	 * @param operation - Operation object.
+	 */
+	public void ringDevice(org.wso2.mdm.agent.beans.Operation operation) {
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 5]
```diff
@@ -54,960 +54,975 @@
...
+	/**
+	 * Wipe the device.
+	 *
+	 * @param operation - Operation object.
+	 */
+	public void wipeDevice(org.wso2.mdm.agent.beans.Operation operation) throws AndroidAgentException {
+		String inputPin;
+		String savedPin =
+				Preference.getString(context,
+						resources.getString(R.string.shared_pref_pin));
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary line breaks

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 6]
```diff
@@ -54,960 +54,975 @@
...
+		}
+	}
+
+	/**
+	 * Display notification.
+	 *
+	 * @param operation - Operation object.
+	 */
+	public void displayNotification(org.wso2.mdm.agent.beans.Operation operation) throws AndroidAgentException {
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 7]
```diff
@@ -54,960 +54,975 @@
...
+						(String) wifiData.get(resources.getString(R.string.intent_extra_password));
+			}
+		} catch (JSONException e) {
+			operation.setStatus(resources.getString(R.string.operation_value_error));
+			resultBuilder.build(operation);
+			throw new AndroidAgentException("Invalid JSON format.", e);
+		}
+
+		WiFiConfig config = new WiFiConfig(context.getApplicationContext());
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 8]
```diff
@@ -54,960 +54,975 @@
...
+		JSONObject result = new JSONObject();
+
+		try {
+			JSONObject wifiData = new JSONObject(operation.getPayLoad().toString());
+			if (!wifiData.isNull(resources.getString(R.string.intent_extra_ssid))) {
+				ssid = (String) wifiData.get(resources.getString(R.string.intent_extra_ssid));
+			}
+			if (!wifiData.isNull(resources.getString(R.string.intent_extra_password))) {
+				password =
+						(String) wifiData.get(resources.getString(R.string.intent_extra_password));
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary line spacing

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 9]
```diff
@@ -54,960 +54,975 @@
...
+		WiFiConfig config = new WiFiConfig(context.getApplicationContext());
+
+		wifiStatus = config.saveWEPConfig(ssid, password);
+
+		try {
+			String status;
+			if (wifiStatus) {
+				status = resources.getString(R.string.shared_pref_default_status);
+				result.put(resources.getString(R.string.operation_status), status);
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 10]
```diff
@@ -54,960 +54,975 @@
...
+		resultBuilder.build(operation);
+	}
+
+	/**
+	 * Disable/Enable device camera.
+	 *
+	 * @param operation - Operation object.
+	 */
+	public void disableCamera(org.wso2.mdm.agent.beans.Operation operation) throws AndroidAgentException {
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 11]
```diff
@@ -54,960 +54,975 @@
...
+	}
+
+	/**
+	 * Install application/bundle.
+	 *
+	 * @param operation - Operation object.
+	 */
+	public void installAppBundle(org.wso2.mdm.agent.beans.Operation operation) throws AndroidAgentException {
+		try {
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 12]
```diff
@@ -54,960 +54,975 @@
...
+				jArray = new JSONArray(operation.getPayLoad().toString());
+				for (int i = 0; i < jArray.length(); i++) {
+					JSONObject appObj = jArray.getJSONObject(i);
+					installApplication(appObj, operation);
+				}
+			}
+			if (Constants.DEBUG_MODE_ENABLED) {
+				Log.d(TAG, "Application bundle installation started");
+			}
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 13]
```diff
@@ -54,960 +54,975 @@
...
+	 * Uninstall application.
+	 *
+	 * @param operation - Operation object.
+	 */
+	public void uninstallApplication(org.wso2.mdm.agent.beans.Operation operation) throws AndroidAgentException {
+		String packageName;
+		String type;
+		try {
+			JSONObject appData = new JSONObject(operation.getPayLoad().toString());
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 14]
```diff
@@ -54,960 +54,975 @@
...
+		}
+	}
+
+	/**
+	 * Encrypt/Decrypt device storage.
+	 *
+	 * @param operation - Operation object.
+	 */
+	public void encryptStorage(org.wso2.mdm.agent.beans.Operation operation) throws AndroidAgentException {
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 15]
```diff
@@ -54,960 +54,975 @@
...
+			Log.e(TAG, msg);
+			throw new AndroidAgentException(msg, e);
+		}
+		operation.setPayLoad(result.toString());
+		operation.setStatus(resources.getString(R.string.operation_value_completed));
+		resultBuilder.build(operation);
+		if (Constants.DEBUG_MODE_ENABLED) {
+			Log.d(TAG, "Encryption process started");
+		}
+
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 16]
```diff
@@ -54,960 +54,975 @@
...
+	 *
+	 * @param operation - Operation object.
+	 */
+	private void muteDevice(org.wso2.mdm.agent.beans.Operation operation) {
+		operation.setStatus(resources.getString(R.string.operation_value_completed));
+		resultBuilder.build(operation);
+
+		AudioManager audioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
+		audioManager.setStreamVolume(AudioManager.STREAM_RING, DEFAULT_VOLUME, DEFAULT_FLAG);
+		if (Constants.DEBUG_MO
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: better to move . sign of the method call to upper line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: better to move . sign of the method call to upper line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: better to move . sign of the method call to upper line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: better to move . sign of the method call to upper line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary line breaks

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary line breaks and spacing

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary line breaks and spacing

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary line breaks and spacing

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: unnecessary empty line

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 1 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 2 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 3 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 4 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 5 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 6 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 8 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 7 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 9 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 10 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 11 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 12 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 13 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 14 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 15 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 16 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Fixed in PR [#70].

**Pull request merged**